test/scapy/layers/l2.uts

```diff
@@ -83,3 +83,27 @@ assert isinstance(p.payload, NoPayload)
 
 p = ARP(pdst='192.168.178.0/24')
 assert "Net" in repr(p)
+
+
+############
+############
++ CookedLinux
+
+= CookedLinux - Basic Dissection
+
+cl = CookedLinux(b'\x00\x00\x03\x04\x00\x06\x00\x00\x00\x00\x00\x00\x6f\x50\x08\x00')
+assert cl.pkttype == 0 # unicast
+assert cl.lladdrtype == 772 # loopback
+assert cl.lladdrlen == 6
+assert cl.src == b'\x00\x00\x00\x00\x00\x00\x6f\x50'
```

Comment:
The `src` field is 8 bytes but address lenght may be e.g. only 6 bytes long. Only 6 bytes from the total 8 bytes should be meaningful ([LINKTYPE_LINUX_SLL](https://www.tcpdump.org/linktypes/LINKTYPE_LINUX_SLL.html) and [LINKTYPE_LINUX_SLL2](https://www.tcpdump.org/linktypes/LINKTYPE_LINUX_SLL2.html)).

Currently existing `CookedLinux` and newly proposed `CookedLinuxV2` use `StrFixedLenField("src", b"", 8)` which always includes all 8 bytes. For example, in this test case the `src` field contains 2 extra bytes that should not be there.  Should this be fixed?  If yes, are there examples of similar field that I could look for how to do this?

Reply:
> Only 6 bytes from the total 8 bytes should be meaningful

This is the case with **some** linklayer addresses, but you could have any length by changing lladdrlen. So I'd say current implementation is fine